ansible/roles/sap-hana-fast-restart/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Determine the number of CPU sockets available in the HANA server
+  shell: |
+    cat /sys/devices/system/node/node*/meminfo | grep MemTotal | awk 'BEGIN {printf "%10s | %20s\n", "NUMA NODE", "MEMORY GB"; while (i++ < 33) printf "-"; printf "\n"} {printf "%10d | %20.3f\n", $2, $4/1048576}'
+  register: cpu_sockets
+
+- name: Read global.ini entries
+  shell: |
+    cat {{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/global.ini
+  register: global_ini
+
+- name: Read indexserver.ini entries
+  shell: |
+    cat {{ sap_hana_usr_mountpoint }}/{{ sap_hana_sid }}/SYS/global/hdb/custom/config/indexserver.ini
+  register: indexserver_ini
+
+- name: Create tmpfs directories for numa nodes
+  file:
+    path: "/hana/tmpfs{{ item }}/{{ sap_hana_sid }}"
+    state: directory
+    owner: "{{ sap_hana_user }}"
+    group: sapsys
+    mode: '777'
+    recurse: yes
+  with_sequence: start=0 end='{{ cpu_sockets.stdout_lines|length - 3 }}'
+  when: global_ini.stdout.find('basepath_persistent_memory_volumes') == -1
```

Comment:
`when` should not be needed here or in the next task as the file and mount modules are idempotent.

Reply:
Done. Pushed the change